sql/expression/like.go

```diff
@@ -0,0 +1,135 @@
+package expression
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
```

Comment:
can you use `https://github.com/src-d/go-mysql-server/tree/master/internal/regex` ?